docs/understand/model.md

```diff
@@ -18,77 +18,69 @@ defines the common concepts of lakeFS.
 
 ## Objects
 
-lakeFS is an object store and borrows concepts from S3.
-
-An _object store_ links objects to paths. An _object_ holds:
-
-* Some _contents_, with unlimited size and format.
-* Some _metadata_, including
-  + _size_ in bytes,
-  + the _creation time_, a timestamp with seconds resolution,
-  + a _checksum_ string which uniquely identifies the contents,
-  + some _user metadata_, a small map of strings to strings.
-
-Similarly to many object stores, lakeFS objects are immutable and never rewritten. They can
-be entirely replaced or deleted, but not modified.
+lakeFS is an interface to manage objects in an object store.
 
 The actual data itself is not stored inside lakeFS directly but in an [underlying object store](#concepts-unique-to-lakefs).
-lakeFS will manage these writes and store a pointer to the object in its metadata database.
+lakeFS manages pointers and additional metadata about these objects. 
 {: .note }
 
 ## Version Control
 
-lakeFS borrows its concepts for version control from Git.
+lakeFS introduces version control semantics for data. Most of these will be familiar to [Git](https://git-scm.com/){:target="_blank"} users:
```

Comment:
I would advice removing the git link and adding more information about the semantics here. it can reduce brain fatigure and context switch for the reader.

Also, instead of introducing, we can use something more empowering like - `lakeFS is spearheading version control semantics for data`. wdyt? @ozkatz

Reply:
Done!